### PR TITLE
diff: Add onwer-uid/gid arguments to diff command

### PR DESCRIPTION
**Pull request by ortogonal**
This makes it possible to use diff on local files were -owner-uid/gid
have been passed to commit, if not diff will say that all local files
are changed.

### CODE REVIEW

File: src/libostree/ostree-diff.h [hunk 1]
```diff
@@ -69,7 +69,9 @@ gboolean ostree_diff_dirs (OstreeDiffFlags flags,
                            GPtrArray      *removed,
                            GPtrArray      *added,
                            GCancellable   *cancellable,
-                           GError        **error);
+                           GError        **error,
```
Comment (cgwalters) [inline src/libostree/ostree-diff.h]: Unfortunately, this is public API, so we can't break it now.  We'd need to add a new version, like `ostree_diff_dirs_with_options()` or something.  Also, the `GError` parameter should be last.

**Comment by cgwalters** [comment]
bot, add author to whitelist

File: src/libostree/ostree-diff.h (hunk 1 above)
Comment (ortogonal) [inline src/libostree/ostree-diff.h]: Ohh! Didn't think of that. I will fix that!

File: src/libostree/libostree.sym [hunk 2]
```diff
@@ -89,6 +89,7 @@ global:
         ostree_deployment_set_index;
         ostree_deployment_set_origin;
         ostree_diff_dirs;
+        ostree_diff_dirs_with_options;
```
Comment (cgwalters) [inline src/libostree/libostree.sym]: This will need to be in the `LIBOSTREE_2017.2` section down below - the section here is for symbols already released.

**Comment by cgwalters** [comment]
The commit series here looks messy; can you rebase on master and squash your patches together?

**Pull request closed without merging**